src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -15,17 +15,290 @@ void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx);
+            if (waterSurface <= effect->pos.y) {
+                effect->type = 0;
+                effect->pos.y = waterSurface;
+                EnZo_Ripple(this, &effect->pos, 0.06f, 0.12f, 200);
+            }
+        }
+        effect++;
+    }
+}
+
+// Splash particles turn into ripples when they hit the surface
+void EnZo_Splash_CreateRipple(EnZo* this) {
+    EnZoEffect* effect;
+    f32 waterSurface;
+    s16 i;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (effect->type == 2) {
+            effect->pos.x += effect->vel.x;
+            effect->pos.y += effect->vel.y;
+            effect->pos.z += effect->vel.z;
+
+            if (effect->vel.y >= -20.0f) {
+                effect->vel.y += effect->vec.y;
+            } else {
+                effect->vel.y = -20.0f;
+                effect->vec.y = 0.0f;
+            }
+            waterSurface = this->actor.posRot.pos.y + this->actor.waterY;
+            if (effect->pos.y < waterSurface) {
+                effect->type = 0;
+                effect->pos.y = waterSurface;
+                EnZo_Ripple(this, &effect->pos, 0.06f, 0.12f, 200);
+            }
+        }
+        effect++;
+    }
+}
+
+void EnZo_DrawRipples(EnZo* this, GlobalContext* globalCtx) {
+    EnZoEffect* effect;
+    s16 i;
+    u8 setup;
+
+    effect = this->effects;
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_zo_eff.c", 217);
+    setup = false;
+    func_80093D84(globalCtx->state.gfxCtx);
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (effect->type == 1) {
+            if (!setup) {
+                do {
+                    gDPPipeSync(POLY_XLU_DISP++);
+                    gSPDisplayList(POLY_XLU_DISP++, D_0600BFC0);
+                    gDPSetEnvColor(POLY_XLU_DISP++, 155, 155, 155, 0);
+                } while (0);
+                setup = true;
+            }
+
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, effect->color.a);
+            Matrix_Translate(effect->pos.x, effect->pos.y, effect->pos.z, MTXMODE_NEW);
+            Matrix_Scale(effect->scale, 1.0f, effect->scale, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_zo_eff.c", 242),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, D_0600C028);
+        }
+        effect++;
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_zo_eff.c", 248);
+}
+
+void EnZo_DrawBubbles(EnZo* this, GlobalContext* globalCtx) {
+    EnZoEffect* effect;
+    s16 i;
+    u8 setup;
+
+    effect = this->effects;
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_zo_eff.c", 260);
+    setup = false;
+    func_80093D84(globalCtx->state.gfxCtx);
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (effect->type == 3) {
+            if (!setup) {
+                do {
+                    gSPDisplayList(POLY_XLU_DISP++, D_060022F0);
```

Comment:
can the do while 0 be on its own line by chance? or does it need to warp the macros? might be worth trying an if 1 in its place or something instead as well